GSoC/2015/Julia-Quantum-:-Framework-for-Solvers.md

```diff
@@ -0,0 +1,183 @@
+ 
+* GitHub Profile : https://github.com/amitjamadagni
+* GSoC 2014 : https://www.google-melange.com/gsoc/project/details/google/gsoc2014/amitjamadagni/5676830073815040
+* GSoC 2014 blog : https://knotsknotted.wordpress.com/category/gsoc/
+* Contribution to Julia : Implementation of ror! and rol! (https://github.com/JuliaLang/julia/pull/9822)         
+* Contribution to Julia Quantum : Normalize to QuArray (https://github.com/JuliaQuantum/QuBase.jl/pull/15)           
+
+---
+### Project Synopsis : The main inspiration is to equip the library with solvers which are on par with QuTiP. 
+Main aims of the project include : 
+
+1. Framework to solve Schrodinger (SE) and Quantum Master Equations (ME) by
+    * Exponential solvers
+    * ODE solvers (RK Methods)
+    * Monte Carlo Wave Function Method (MCWF)
+    * Steady State Methods
+
+2. Parallelizing some of the above tasks (Integrating QuArrays and methods involved in MCWF) using the existing infrastructure for parallel computing features in Julia to achieve greater efficiency.
+
+3. In the course of integrating other packages, contribute back to improve their functionality and usability.
+
+---
+## Details :
+
+### Implementation :
+The implementation details can be broadly classified into pre and post parallelization. The pre parallelization phase would mainly deal with getting the basic algorithms and design work keeping in mind the parallelization that could be achieved. Once the base is set, I hope to work on parallelizing the implementation. Hamiltonian and the Master equation are used for propagation (Most of the theoretical  reference is from the QuTiP Documentation page [here] (http://qutip.org/docs/3.1.0/guide/guide-dynamics.html) and Python notebooks [here] (http://nbviewer.ipython.org/github/qutip/qutip-notebooks/tree/master/examples/) ). The main algorithmic construction is of the methods used for state evolution. 
+
+* Exponential Solvers : The implemenation would use packages like [ExpmV.jl] (https://github.com/marcusps/ExpmV.jl) and [Expokit.jl] (https://github.com/acroy/Expokit.jl) to implement the evolution.
+
+* Differential Equation Solvers : The implementation would use the solvers from [ODE.jl] (https://github.com/JuliaLang/ODE.jl) package.
+
+* Monte Carle Wavefunction : The implementation would be based on the proposed implementation in QuDOS.jl package and also with reference from QuTiP.
+
+* Steady State equation : Various methods based out of QuTiP. 
+
+The parallelization techniques can be achieved at certain level of abstraction. Some of the ideas are as follows :
+
+*  Level of Abstraction --> Algorithmic (for example : Monte Carlo methods)
+
+*  Level of Abstraction --> Type definitions (for example : `QuArray` based out of DArray or using the package [ParallelSparseMatMul.jl] (https://github.com/madeleineudell/ParallelSparseMatMul.jl) given the fact `QuArray`s are based on arrays.
+
```

Comment:
Should we have 1-2 sentences here to summarize what we will have at the end of the project? Something like

At the end of the project, JuliaQuantum will be equipped with a framework to solve dynamical equations arising in quantum mechanics. The framework itself and the structures defined in QuBase will be useable with distributed/shared arrays to enable development of methods using parallel computing.